heaps/min_heap.py

```diff
@@ -19,21 +18,40 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: log n
+            Space Complexity: 1
         """
-        pass
+        if value == None:
+            value = key
+        
+        #make a new node
+        new_node = HeapNode(key, value)
+        #add in the new node
+        self.store.append(new_node)
+        #figure out what the very last index is because that is where we added to
+        index = len(self.store) - 1
+        #shift and restore the heap
+        self.heap_up(index)
+        
 
     def remove(self):
         """ This method removes and returns an element from the heap
             maintaining the heap structure
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: log n
+            Space Complexity: 1
```

Comment:
✨ Nice, however because of the recursive call stack of `heap_down`, space complexity is also O(log n)